[TEST] reference/mapping/types/date_nanos failing

Build scan (7.6): https://gradle-enterprise.elastic.co/s/f3d6mbfyo554w/tests/jbm2x6tnwtxku-ydwigczazxxf6

```
REPRODUCE WITH: ./gradlew ':docs:integTestRunner' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/mapping/types/date_nanos/line_32}" -Dtests.seed=50B1899A04B6F25D -Dtests.security.manager=true -Dtests.locale=en-AU -Dtests.timezone=PNT -Dcompiler.java=13
```

```
java.lang.AssertionError: Failure at [reference/mapping/types/date_nanos:58]: expected [2xx] status code but api [raw[method=GET path=my_index/_search]] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"script_exception","reason":"runtime error","script_stack":["doc['date'].date.nanos","           ^---- HERE"],"script":"doc['date'].date.nanos","lang":"painless","stack_trace":"ScriptException[runtime error]; nested: IllegalArgumentException[Illegal list shortcut value [date].];\n\tat org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:94)\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:17)\n\tat org.elasticsearch.search.fetch.subphase.ScriptFieldsFetchSubPhase.hitsExecute(ScriptFieldsFetchSubPhase.java:67)\n\tat org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:178)\n\tat org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:387)\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:367)\n\tat org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:343)\n\tat org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)\n\tat org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58)\n\tat org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\nCaused by: java.lang.IllegalArgumentException: Illegal list shortcut value [date].\n\tat org.elasticsearch.painless.Def.lookupGetter(Def.java:392)\n\tat org.elasticsearch.painless.DefBootstrap$PIC.lookup(DefBootstrap.java:153)\n\tat org.elasticsearch.painless.DefBootstrap$PIC.fallback(DefBootstrap.java:207)\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:12)\n\t... 16 more\n"}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":"my_index","node":"hf4_iqIZQiuWYkrd_VbwRg","reason":{"type":"script_exception","reason":"runtime error","script_stack":["doc['date'].date.nanos","           ^---- HERE"],"script":"doc['date'].date.nanos","lang":"painless","caused_by":{"type":"illegal_argument_exception","reason":"Illegal list shortcut value [date].","stack_trace":"java.lang.IllegalArgumentException: Illegal list shortcut value [date].\n\tat org.elasticsearch.painless.Def.lookupGetter(Def.java:392)\n\tat org.elasticsearch.painless.DefBootstrap$PIC.lookup(DefBootstrap.java:153)\n\tat org.elasticsearch.painless.DefBootstrap$PIC.fallback(DefBootstrap.java:207)\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:12)\n\tat org.elasticsearch.search.fetch.subphase.ScriptFieldsFetchSubPhase.hitsExecute(ScriptFieldsFetchSubPhase.java:67)\n\tat org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:178)\n\tat org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:387)\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:367)\n\tat org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:343)\n\tat org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)\n\tat org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58)\n\tat org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\n"},"stack_trace":"ScriptException[runtime error]; nested: IllegalArgumentException[Illegal list shortcut value [date].];\n\tat org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:94)\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:17)\n\tat org.elasticsearch.search.fetch.subphase.ScriptFieldsFetchSubPhase.hitsExecute(ScriptFieldsFetchSubPhase.java:67)\n\tat org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:178)\n\tat org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:387)\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:367)\n\tat org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:343)\n\tat org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)\n\tat org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58)\n\tat org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\nCaused by: java.lang.IllegalArgumentException: Illegal list shortcut value [date].\n\tat org.elasticsearch.painless.Def.lookupGetter(Def.java:392)\n\tat org.elasticsearch.painless.DefBootstrap$PIC.lookup(DefBootstrap.java:153)\n\tat org.elasticsearch.painless.DefBootstrap$PIC.fallback(DefBootstrap.java:207)\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:12)\n\t... 16 more\n"}}],"stack_trace":"Failed to execute phase [query], all shards failed; shardFailures {[hf4_iqIZQiuWYkrd_VbwRg][my_index][0]: RemoteTransportException[[node-0][127.0.0.1:46007][indices:data/read/search[phase/query]]]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Illegal list shortcut value [date].]; }\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:545)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:306)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:574)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:386)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.access$200(AbstractSearchAsyncAction.java:66)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:242)\n\tat org.elasticsearch.action.search.SearchExecutionStatsCollector.onFailure(SearchExecutionStatsCollector.java:73)\n\tat org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:59)\n\tat org.elasticsearch.action.search.SearchTransportService$ConnectionCountingHandler.handleException(SearchTransportService.java:423)\n\tat org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1118)\n\tat org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1227)\n\tat org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1201)\n\tat org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:60)\n\tat org.elasticsearch.action.support.ChannelActionListener.onFailure(ChannelActionListener.java:56)\n\tat org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:65)\n\tat org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58)\n\tat org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)\n\tat java.lang.Thread.run(Thread.java:748)\nCaused by: ScriptException[runtime error]; nested: IllegalArgumentException[Illegal list shortcut value [date].];\n\tat org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:94)\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:17)\n\tat org.elasticsearch.search.fetch.subphase.ScriptFieldsFetchSubPhase.hitsExecute(ScriptFieldsFetchSubPhase.java:67)\n\tat org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:178)\n\tat org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:387)\n\tat org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:367)\n\tat org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:343)\n\tat org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)\n\t... 9 more\nCaused by: java.lang.IllegalArgumentException: Illegal list shortcut value [date].\n\tat org.elasticsearch.painless.Def.lookupGetter(Def.java:392)\n\tat org.elasticsearch.painless.DefBootstrap$PIC.lookup(DefBootstrap.java:153)\n\tat org.elasticsearch.painless.DefBootstrap$PIC.fallback(DefBootstrap.java:207)\n\tat org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:12)\n\t... 16 more\n"},"status":400}]


[2020-02-05T03:41:11,784][INFO ][o.e.s.DocsClientYamlTestSuiteIT] [test] Stash dump on test failure [{
  "stash" : {
    "body" : {
      "error" : {
        "root_cause" : [
          {
            "type" : "script_exception",
            "reason" : "runtime error",
            "script_stack" : [
              "doc['date'].date.nanos",
              "           ^---- HERE"
            ],
            "script" : "doc['date'].date.nanos",
            "lang" : "painless",
            "stack_trace" : "ScriptException[runtime error]; nested: IllegalArgumentException[Illegal list shortcut value [date].];
	at org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:94)
	at org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:17)
	at org.elasticsearch.search.fetch.subphase.ScriptFieldsFetchSubPhase.hitsExecute(ScriptFieldsFetchSubPhase.java:67)
	at org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:178)
	at org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:387)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:367)
	at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:343)
	at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146)
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)
	at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58)
	at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Illegal list shortcut value [date].
	at org.elasticsearch.painless.Def.lookupGetter(Def.java:392)
	at org.elasticsearch.painless.DefBootstrap$PIC.lookup(DefBootstrap.java:153)
	at org.elasticsearch.painless.DefBootstrap$PIC.fallback(DefBootstrap.java:207)
	at org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:12)
	... 16 more
"
          }
        ],
        "type" : "search_phase_execution_exception",
        "reason" : "all shards failed",
        "phase" : "query",
        "grouped" : true,
        "failed_shards" : [
          {
            "shard" : 0,
            "index" : "my_index",
            "node" : "hf4_iqIZQiuWYkrd_VbwRg",
            "reason" : {
              "type" : "script_exception",
              "reason" : "runtime error",
              "script_stack" : [
                "doc['date'].date.nanos",
                "           ^---- HERE"
              ],
              "script" : "doc['date'].date.nanos",
              "lang" : "painless",
              "caused_by" : {
                "type" : "illegal_argument_exception",
                "reason" : "Illegal list shortcut value [date].",
                "stack_trace" : "java.lang.IllegalArgumentException: Illegal list shortcut value [date].
	at org.elasticsearch.painless.Def.lookupGetter(Def.java:392)
	at org.elasticsearch.painless.DefBootstrap$PIC.lookup(DefBootstrap.java:153)
	at org.elasticsearch.painless.DefBootstrap$PIC.fallback(DefBootstrap.java:207)
	at org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:12)
	at org.elasticsearch.search.fetch.subphase.ScriptFieldsFetchSubPhase.hitsExecute(ScriptFieldsFetchSubPhase.java:67)
	at org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:178)
	at org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:387)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:367)
	at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:343)
	at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146)
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)
	at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58)
	at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
"
              },
              "stack_trace" : "ScriptException[runtime error]; nested: IllegalArgumentException[Illegal list shortcut value [date].];
	at org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:94)
	at org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:17)
	at org.elasticsearch.search.fetch.subphase.ScriptFieldsFetchSubPhase.hitsExecute(ScriptFieldsFetchSubPhase.java:67)
	at org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:178)
	at org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:387)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:367)
	at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:343)
	at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146)
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)
	at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58)
	at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: Illegal list shortcut value [date].
	at org.elasticsearch.painless.Def.lookupGetter(Def.java:392)
	at org.elasticsearch.painless.DefBootstrap$PIC.lookup(DefBootstrap.java:153)
	at org.elasticsearch.painless.DefBootstrap$PIC.fallback(DefBootstrap.java:207)
	at org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:12)
	... 16 more
"
            }
          }
        ],
        "stack_trace" : "Failed to execute phase [query], all shards failed; shardFailures {[hf4_iqIZQiuWYkrd_VbwRg][my_index][0]: RemoteTransportException[[node-0][127.0.0.1:46007][indices:data/read/search[phase/query]]]; nested: ScriptException[runtime error]; nested: IllegalArgumentException[Illegal list shortcut value [date].]; }
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:545)
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:306)
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:574)
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:386)
	at org.elasticsearch.action.search.AbstractSearchAsyncAction.access$200(AbstractSearchAsyncAction.java:66)
	at org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:242)
	at org.elasticsearch.action.search.SearchExecutionStatsCollector.onFailure(SearchExecutionStatsCollector.java:73)
	at org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:59)
	at org.elasticsearch.action.search.SearchTransportService$ConnectionCountingHandler.handleException(SearchTransportService.java:423)
	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1118)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1227)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1201)
	at org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:60)
	at org.elasticsearch.action.support.ChannelActionListener.onFailure(ChannelActionListener.java:56)
	at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71)
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:65)
	at org.elasticsearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:58)
	at org.elasticsearch.action.ActionRunnable$2.doRun(ActionRunnable.java:73)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:692)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: ScriptException[runtime error]; nested: IllegalArgumentException[Illegal list shortcut value [date].];
	at org.elasticsearch.painless.PainlessScript.convertToScriptException(PainlessScript.java:94)
	at org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:17)
	at org.elasticsearch.search.fetch.subphase.ScriptFieldsFetchSubPhase.hitsExecute(ScriptFieldsFetchSubPhase.java:67)
	at org.elasticsearch.search.fetch.FetchPhase.execute(FetchPhase.java:178)
	at org.elasticsearch.search.SearchService.executeFetchPhase(SearchService.java:387)
	at org.elasticsearch.search.SearchService.executeQueryPhase(SearchService.java:367)
	at org.elasticsearch.search.SearchService.lambda$executeQueryPhase$1(SearchService.java:343)
	at org.elasticsearch.action.ActionListener.lambda$map$2(ActionListener.java:146)
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)
	... 9 more
Caused by: java.lang.IllegalArgumentException: Illegal list shortcut value [date].
	at org.elasticsearch.painless.Def.lookupGetter(Def.java:392)
	at org.elasticsearch.painless.DefBootstrap$PIC.lookup(DefBootstrap.java:153)
	at org.elasticsearch.painless.DefBootstrap$PIC.fallback(DefBootstrap.java:207)
	at org.elasticsearch.painless.PainlessScript$Script.execute(doc['date'].date.nanos:12)
	... 16 more
"
      },
      "status" : 400
    }
  }
}]
```

--- a/docs/reference/mapping/types/date_nanos.asciidoc
+++ b/docs/reference/mapping/types/date_nanos.asciidoc
@@ -61,7 +61,7 @@ GET my_index/_search
     "my_field" : {
       "script" : {
         "lang" : "painless",
-        "source" : "doc['date'].date.nanos" <6>
+        "source" : "doc['date'].value.nano" <6>
       }
     }
   }

